github.com/cockroachdb/pebble/internal/metamorphic: TestMeta failed

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/12042175?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/12042175?buildTab=artifacts#meta) on refs/heads/master @ [b013ca78e9dc](https://github.com/cockroachdb/pebble/commits/b013ca78e9dc):


```
		===== SEED =====
		1696414063505387527
		===== DIFF =====
		/artifacts/meta/231004-100743.5053630517238/{standard-000,random-010}
		@@ -2890,11 +2890,11 @@
 db.Delete("tyuryzsp@5") // <nil> #2889
 db.SingleDelete("srjlbmj@10", false /* maybeReplaceDelete */) // <nil> #2890
 iter29.SeekGE("lulnkxbbg@4", "") // [false] <nil> #2891
 iter29.SeekPrefixGE("busgvdi") // [false] pebble: SeekPrefixGE supplied with key outside of lower bound #2892
 iter30.SetOptions("", "", 0 /* key types */, 0, 0, false /* use L6 filters */, "" /* masking suffix */) // <nil> #2893
-iter30.SeekGE("axdcuzcej@8", "") // [true,"axdcuzcej@8","nszffnwdcqrf",<no range>] <nil> #2894
+iter30.SeekGE("axdcuzcej@8", "") // [true,"azlgnktrlb","zrcvliuxszrm",<no range>] <nil> #2894
 iter28.Next("") // [true,"wpncrvroqxy",<no point>,["wpncrvroqxy","wrkhdbhhhan")=>{"@8"="vk"}*] <nil> #2895
 iter28.First() // [true,"wpncrvroqxy",<no point>,["wpncrvroqxy","wrkhdbhhhan")=>{"@8"="vk"}] <nil> #2896
 iter28.SeekPrefixGE("ldmstodov@3") // [false] pebble: SeekPrefixGE supplied with key outside of lower bound #2897
 iter28.SetBounds("cuhxs", "dhdqzz") // <nil> #2898
 iter28.SeekLT("dhdqzz", "") // [false] <nil> #2899

```
<details><summary>Help</summary>
<p>
To reproduce, try:

```bash
go test -tags 'invariants' -exec 'stress -p 1' -timeout 0 -test.v -run TestMeta$ ./internal/metamorphic -seed 1696414063505387527 -ops "uniform:5000-10000"
```
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/12059359?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_Pebble_Metamorphic/12059359?buildTab=artifacts#meta) on refs/heads/master @ [d4bf20c546d3](https://github.com/cockroachdb/pebble/commits/d4bf20c546d3):


```
=== RUN   TestMeta
--- FAIL: TestMeta (7.11s)
=== RUN   TestMeta/execution
    --- FAIL: TestMeta/execution (0.00s)
=== RUN   TestMeta/execution/random-017
=== PAUSE TestMeta/execution/random-017
=== CONT  TestMeta/execution/random-017
        --- FAIL: TestMeta/execution/random-017 (0.65s)
```
<details><summary>Help</summary>
<p>
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

github.com/cockroachdb/pebble/internal/metamorphic.TestMeta [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicRace/12059360?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicRace/12059360?buildTab=artifacts#meta) on refs/heads/master @ [d4bf20c546d3](https://github.com/cockroachdb/pebble/commits/d4bf20c546d3):


```
=== RUN   TestMeta
--- FAIL: TestMeta (76.14s)
=== RUN   TestMeta/execution
    --- FAIL: TestMeta/execution (0.00s)
=== RUN   TestMeta/execution/random-021
=== PAUSE TestMeta/execution/random-021
=== CONT  TestMeta/execution/random-021
        --- FAIL: TestMeta/execution/random-021 (4.46s)
```
<details><summary>Help</summary>
<p>
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMeta.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/metamorphic/generator.go
+++ b/metamorphic/generator.go
@@ -737,13 +737,13 @@ func (g *generator) newIterUsingClone() {
 		refreshBatch = g.rng.Intn(2) == 1
 	}
 
-	var opts iterOpts
+	opts := g.itersLastOpts[existingIterID]
+	// With 50% probability, consider modifying the iterator options used by the
+	// clone.
 	if g.rng.Intn(2) == 1 {
 		g.maybeMutateOptions(readerID, &opts)
-		g.itersLastOpts[iterID] = opts
-	} else {
-		g.itersLastOpts[iterID] = g.itersLastOpts[existingIterID]
 	}
+	g.itersLastOpts[iterID] = opts
 
 	g.iterCreationTimestamp[iterID] = g.keyManager.nextMetaTimestamp()
 	g.iterReaderID[iterID] = g.iterReaderID[existingIterID]

